src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,805 @@
+#
+# Conjugation
+#
+########################################################################
+
+function Base.show(io::IO, x::GroupConjClass{T,S}) where T <: MatrixGroup where S <: MatrixGroupElem
+  show(x.repr)
+  print(" ^ ")
+  show(x.X)
+end
+
+function Base.show(io::IO, x::GroupConjClass{T,S}) where T <: MatrixGroup where S <: MatrixGroup
+  show(x.repr)
+  print(" ^ ")
+  show(x.X)
+end
+
+function Base.:^(H::MatrixGroup, y::MatrixGroupElem)
+   if isdefined(H,:gens)
+      K = MatrixGroup(H.deg, H.ring)
+      K.gens = [y^-1*x*y for x in H.gens]
+      return K
+   else
+      return MatrixGroup(H.deg,H.ring,H.X^y.X)
+   end
+end
+
+function conjugacy_classes_subgroups(G::MatrixGroup)
+   L=GAP.gap_to_julia(Vector{GapObj},GAP.Globals.ConjugacyClassesSubgroups(G.X))
+   return GroupConjClass{typeof(G), typeof(G)}[ _conjugacy_class(G,MatrixGroup(G.deg,G.ring,GAP.Globals.Representative(cc)),cc) for cc in L]
+end
+
+function conjugacy_classes_maximal_subgroups(G::MatrixGroup)
+  L = GAP.gap_to_julia(Vector{GapObj},GAP.Globals.ConjugacyClassesMaximalSubgroups(G.X))
+   return GroupConjClass{typeof(G), typeof(G)}[ _conjugacy_class(G,MatrixGroup(G.deg,G.ring,GAP.Globals.Representative(cc)),cc) for cc in L]
+end
+
+function Base.rand(C::GroupConjClass{S,T}) where S<:MatrixGroup where T<:MatrixGroup
+   return MatrixGroup(C.X.deg,C.X.ring,GAP.Globals.Random(C.CC))
+end
+
+function elements(C::GroupConjClass{S, T}) where S where T<:MatrixGroup
+   L=GAP.Globals.AsList(C.CC)
+   l = Vector{T}(undef, length(L))
+   for i in 1:length(l)
+      l[i] = _as_subgroup(C.X, L[i])
+   end
+   return l
+end
+
+
+
+
+
+#=
+
+########################################################################
+#
+# conversions     (this should be no longer necessary, and it's also wrong)
+#
+########################################################################
+#TODO: support other field types
+
+function FieldGapToOscar(F::GapObj)
+   p = GAP.Globals.Characteristic(F)
+   q = GAP.Globals.Size(F)
+   d = Base.Integer(log(p,q))
+
+   return GF(p,d)
+end
+
+function FieldOscarToGap(F::FqNmodFiniteField)
+   p = Int64(characteristic(F))
+   d = Int64(degree(F))
+
+   return GAP.Globals.GF(p,d)
+end
+
+function FieldElemGapToOscar(x::FFE, Fgap::GapObj, F::FqNmodFiniteField)
+   q = GAP.Globals.Size(Fgap)
+   z = gen(F)
+   if GAP.Globals.IsZero(x)
+      return 0*z
+   else
+      d = Integer(GAP.Globals.LogFFE(x, GAP.Globals.Z(q)))
+      return z^d
+   end
+end
+
+function FieldElemGapToOscar(x::FFE, F::GapObj)
+   q = GAP.Globals.Size(F)
+   K,z = FieldGapToOscar(F)
+   if GAP.Globals.IsZero(x)
+      return 0*z
+   else
+      d = Integer(GAP.Globals.LogFFE(x, GAP.Globals.Z(q)))
+      return z^d
+   end
+end
+
+function FieldElemOscarToGap(x::fq_nmod, F::FqNmodFiniteField)
+   q = Int64(order(F))
+   d = degree(F)
+   z = gen(F)
+   v = [Int64(coeff(x,i)) for i in 0:d]
+   y = sum([GAP.Globals.Z(q)^i*v[i+1] for i in 0:d])
+   
+   return y
+end
+
```

Comment:
Conversions to GAP will in general need more context. In the case of finite field elements that can be constructed using `Z(q)`, the situation is easy, but for example algebraic extensions in characteristic zero will require that the field on the GAP side gets created just once, and then this GAP object must be used throughout the whole session; this means that the GAP field (or some object that describes the conversion) has to be cached somewhere, such that it can be fetched when the Oscar field is given; perhaps the same is useful in the other direction as well.